source/MaterialXGenShader/Nodes/SourceCodeNode.cpp

```diff
@@ -81,13 +81,44 @@ void SourceCodeNode::emitFunctionCall(const ShaderNode& node, GenContext& contex
         static const string prefix("{{");
         static const string postfix("}}");
 
-        // Inline expressions can only have a single output
+        size_t pos = 0;
```

Comment:
A bit different then what we discussed as you perform two loops on the inline but this may be clearer to read.

Reply:
Yeah I agree it would be nice to avoid parsing the text twice. Maybe you could just push each snippet of text into a string vector, replacing any {{foo}} input arguments by it's connected variable name or it's local value variable if it's unconnected (at the same time you emit the local variables where needed). And then in the second pass you just emit all of these stored text snippets, constructing the final expression.